CalculatorLibrary/CalculatorLibrary.cs

```diff
@@ -0,0 +1,196 @@
+
+                    while (!double.TryParse(num2, out cleanNum2))
+                    {
+                        Console.Write("This is not valid input. Please enter an integer value: ");
+                        num2 = Console.ReadLine();
+                    }
+                    
+                    if (Regex.IsMatch(op, "[s|p|l]"))
+                    {
+                        result = AdvancedOperations(cleanNum1, cleanNum2, op);
+                        switch (op)
+                        {
+                            case "s":
+                                result = Math.Sqrt(cleanNum1);
+                                Console.WriteLine($"{cleanNum1} square rooted = {result}");
+                                break;
+                            case "p":
+                                result = Math.Pow(cleanNum1, cleanNum2);
+                                Console.WriteLine($"{cleanNum1} to the power of {cleanNum2} = {result}");
+                                break;
+                            case "l":
+                                result = Math.Log(cleanNum1, cleanNum2);
+                                Console.WriteLine($"Logarithm {cleanNum1} with the base of {cleanNum2} = {result}");
+                                break;
+                            default:
+                                break;
+                        }
+                    }
+                    else
+                    {
+                        result = BasicOperations(cleanNum1, cleanNum2, op);
+                        Console.WriteLine($"{cleanNum1} {op} {cleanNum2} = {result}");
+                        Console.WriteLine("Your result: {0:0.##}\n", result);
+                    }
+                }
+                if (string.Equals(op, "delete"))
+                {
+                    operations.Clear();
+                }
+            }
+        }
+
+        public double BasicOperations(double num1, double num2, string op)
+        {
+            double result = double.NaN;
+            writer.WriteStartObject();
+            writer.WritePropertyName("Operand1");
+            writer.WriteValue(num1);
+            writer.WritePropertyName("Operand2");
+            writer.WriteValue(num2);
+            writer.WritePropertyName("Operation");
+
+            switch (op)
+            {
+                case "+":
+                    result = num1 + num2;
+                    writer.WriteValue("Add");
+                    break;
+                case "-":
+                    result = num1 - num2;
+                    writer.WriteValue("Substract");
+                    break;
+                case "*":
+                    result = num1 * num2;
+                    writer.WriteValue("Multiply");
+                    break;           
+                case "/":
+                    if (num2 != 0)
+                    {
+                        result = num1 / num2;
+                        writer.WriteValue("Divide");
```

Comment:
As per the tutorial this is incorrect and will result in the wrong error 😄 .

Test and see for yourself 😱 !

After you have tested dividing by zero please update as per the microsoft documentation:
https://learn.microsoft.com/en-us/visualstudio/get-started/csharp/tutorial-console-part-2?view=vs-2022